UTAT_Compression_Algorithm/C/compression.c

```diff
@@ -0,0 +1,67 @@
+#include <gsl/gsl_matrix.h>
+#include <gsl/gsl_vector.h>
+#include <gsl/gsl_blas.h>
+#include <math.h>
+
+#include "helperlib.h"
+
+#define dynamic_range 32
+#define weight_resolution 4
+#define s_mid 0
+#define register_size 50
+
+/*
+Calculates the local sum at a given x,y point in the dataset
+Input: x-coordinate, y-coordinate, total number of x-coordinates, and the band of the dataset to calculate the local sum
+*/ 
+double localSum(int x, int y, int Nx, gsl_matrix* data){
+    double localSum = 0;
+    if (y == 0 && x > 0){
+        localSum = 4*gsl_matrix_get(data,y,x-1);
```

Comment:
is there a GSL function that handles array out of bounds exceptions? Perhaps it's handled in the `gsl_matrix_get` function itself. However if it's not handled there, we should have an explicit handling before we call the get function.